holidays/countries/thailand.py

```diff
@@ -12,156 +12,885 @@
+        elif 1989 <= year <= 1997:
+            self[date(year, APR, 12)] = songkran_festival_en
+            self[date(year, APR, 13)] = songkran_festival_en
+            self[date(year, APR, 14)] = songkran_festival_en
+        # 1998-Present, celebrated on Apr 13-15
+        # (Except for 2020 due to Covid-19 outbreaks)
+        elif year >= 1998 and year != 2020:
+            self[date(year, APR, 13)] = songkran_festival_en
+            self[date(year, APR, 14)] = songkran_festival_en
+            self[date(year, APR, 15)] = songkran_festival_en
+
+        # !!! Songkran Festival (in lieu) !!!
+        # วันหยุดชดเชยวันสงกรานต์
+        # If Songkran happened to be held on the weekends, only one in-lieu
+        #   public holiday is added, No in lieus for SUN-MON-TUE case
+        # Status: In Use
+        # Sources:
+        #   https://github.com/dr-prodigy/python-holidays/pull/929
+        songkran_festival_in_lieu_en = "Songkran Festival (in lieu)"
+
+        # Applied Automatically for Monday if on Weekends: 1961-1973
+        # No In Lieu days available: 1974-1988
+        # Case-by-Case application for Workday if on Weekends: 1989-1994
+        # Applied Automatically for Workday if on Weekends: 1995-1997
+        # Case-by-Case application for Workday if on Weekends: 1998-2000
+        # Applied Automatically for Workday if on Weekends: 2001-Present
+        if self.observed and (1995 <= year <= 1997 or year >= 2001):
+            dt = date(year, APR, 15) if year >= 2001 else date(year, APR, 14)
+            # CASE 1: THU-FRI-SAT -> 1 in-lieu on MON
+            if dt.weekday() == SAT:
+                self[dt + td(days=+2)] = songkran_festival_in_lieu_en
+            # CASE 2: FRI-SAT-SUN -> 1 in-lieu on MON
+            elif dt.weekday() == SUN:
+                self[dt + td(days=+1)] = songkran_festival_in_lieu_en
+            # CASE 3: SAT-SUN-MON -> 1 in-lieu on TUE
+            elif dt.weekday() == MON:
+                self[dt + td(days=+1)] = songkran_festival_in_lieu_en
+
+        # !!! Labour day !!!
+        # วันแรงงานแห่งชาติ
+        # Status: In-Use
+        # Starts in the present form in 1974 (B.E. 2517)
+        # Does existed officially since 1956 (B.E. 2499),
+        #   but wasn't a public holiday until then
+        # *** NOTE: only observed by financial and private sectors
+        # Sources:
+        #   https://www.thairath.co.th/lifestyle/culture/1832869
+        national_labour_day_en = "National Labour Day"
+
+        if year >= 1974:
+            add_holiday(date(year, MAY, 1), national_labour_day_en)
+
+        # !!! National Day (24 June) !!!
+        # วันชาติ
+        # Status: Defunct (Historical)
+        # Starts in 1939 (B.E. 2482) by Plaek Phibunsongkhram
+        # Replaced by Rama IX's birthday in 1960 (B.E. 2503) by Sarit Thanarat
+        # Sources:
+        #   (Ministry of Culture 's wbm) http://tiny.cc/wa_mincul_nat_day
+        national_day_khana_ratsadon_en = "National Day"
```

Comment:
Uh, I would prefer to avoid in-lines here as the naming tended to get changed around a bit for one reason or another. I wouldn't mind shorten the string variable though to keep the logic part clear and in-line :)

Reply:
@KJhellico @arkid15r 
Is the following Ukranian method workable?
```python
name = (
     "День захисників і захисниць України"
     if year >= 2021
     else "День захисника України"
)
```

Reply:
Nevermind, I think I misunderstood, gimme a minute or so to get this implemented